src/SLCore/SLCoreHandle.cs

```diff
@@ -0,0 +1,125 @@
+
+namespace SonarLint.VisualStudio.SLCore;
+
+internal sealed class SLCoreHandle : IDisposable
+{
+    private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+    private readonly ISLCoreRpcFactory slCoreRpcFactory;
+    private readonly IServerConnectionsProvider serverConnectionConfigurationProvider;
+    private readonly IConfigScopeUpdater configScopeUpdater;
+    private readonly ISLCoreConstantsProvider constantsProvider;
+    private readonly ISLCoreFoldersProvider slCoreFoldersProvider;
+    private readonly ISLCoreEmbeddedPluginJarLocator slCoreEmbeddedPluginJarProvider;
+    private readonly IThreadHandling threadHandling;
+    private ISLCoreRpc slCoreRpc;
+
+
+    public SLCoreHandle(ISLCoreRpcFactory slCoreRpcFactory, ISLCoreConstantsProvider constantsProvider, ISLCoreFoldersProvider slCoreFoldersProvider,
+        IServerConnectionsProvider serverConnectionConfigurationProvider, ISLCoreEmbeddedPluginJarLocator slCoreEmbeddedPluginJarProvider,
+        IActiveSolutionBoundTracker activeSolutionBoundTracker, IConfigScopeUpdater configScopeUpdater, IThreadHandling threadHandling)
+    {
+        this.slCoreRpcFactory = slCoreRpcFactory;
+        this.constantsProvider = constantsProvider;
+        this.slCoreFoldersProvider = slCoreFoldersProvider;
+        this.serverConnectionConfigurationProvider = serverConnectionConfigurationProvider;
+        this.slCoreEmbeddedPluginJarProvider = slCoreEmbeddedPluginJarProvider;
+        this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+        this.configScopeUpdater = configScopeUpdater;
+        this.threadHandling = threadHandling;
+    }
+
+    public async Task InitializeAsync()
+    {
+        threadHandling.ThrowIfOnUIThread();
+        
+        slCoreRpc = slCoreRpcFactory.StartNewRpcInstance();
+
+        if (!slCoreRpc.ServiceProvider.TryGetTransientService(out ILifecycleManagementSLCoreService lifecycleManagementSlCoreService) ||
+            !slCoreRpc.ServiceProvider.TryGetTransientService(out ITelemetrySLCoreService telemetrySlCoreService))
+        {
+            throw new InvalidOperationException(SLCoreStrings.ServiceProviderNotInitialized);
+        }
+
+        var serverConnectionConfigurations = serverConnectionConfigurationProvider.GetServerConnections();
+        var (storageRoot, workDir, sonarlintUserHome) = slCoreFoldersProvider.GetWorkFolders();
+
+        await lifecycleManagementSlCoreService.InitializeAsync(new InitializeParams(
+            constantsProvider.ClientConstants,
+            new HttpConfigurationDto(new SslConfigurationDto()),
+            constantsProvider.FeatureFlags,
+            storageRoot,
+            workDir,
+            embeddedPluginPaths: slCoreEmbeddedPluginJarProvider.ListJarFiles(),
+            connectedModeEmbeddedPluginPathsByKey: new Dictionary<string, string>(),
+            enabledLanguagesInStandaloneMode: new List<Language>
+            {
```

Comment:
should this be made configurable?

Reply:
@ugras-ergun-sonarsource

Reply:
for now, it's OK I think it's easy to maintain at this stage. We can reconsider if this causes pain.